src/db/index.js

```diff
@@ -200,4 +199,50 @@ export default class {
                 .delete();
         }
     }
+
+    async downloadDb() {
+        await this.openDB();
+        const rc = {};
+        try {
+            for (const table of this.db.tables) {
+                let schema = table.schema.indexes.reduce(
+                    (acc, index) => acc + `,${index.src}`,
+                    ''
+                );
+                schema = table.schema.primKey.src + schema;
+                rc[table.name] = {
+                    schema: schema,
+                    data: await table.toArray(),
+                };
+            }
+        } catch (err) {
+            err.message = `Unable to download ${this.db.name}.
+            ${err.message}
+            `;
+            throw err;
+        }
+        return rc;
+    }
+
+    async uploadDb(bulk) {
```

Comment:
s/uploadDb/upload/

Reply:
Not exactly sure what you mean.

Reply:
I think he means change `uploadDb(bulk)` to `upload(bulk)`